capnp/src/message.rs

```diff
@@ -206,14 +220,38 @@ impl ReaderSegments for SegmentArray<'_> {
     }
 }
 
-impl ReaderSegments for [&[u8]] {
+impl<I> ReaderSegments for [I]
+where
+    I: AsRef<[u8]>,
+{
+    fn get_segment(&self, id: u32) -> Option<&[u8]> {
+        self.get(id as usize).map(|i| i.as_ref())
+    }
+
+    fn len(&self) -> usize {
+        self.len()
+    }
+
+    fn is_empty(&self) -> bool {
+        self.is_empty()
+    }
+}
+
+impl<I> ReaderSegments for Vec<I>
```

Comment:
To make CI pass, I think you need to hide this behind `#[cfg(feature = "alloc")]` and also fully namespace `Vec` as `alloc::vec::Vec`.

Reply:
Thank you very much for the hint! The fixes seem to have worked.